hw3/02-url.js

```diff
@@ -1,42 +1,59 @@
 const http = require('http');
 
 const port = process.env.PORT || 5000;
-
+/*
+In this exercise, you will parse the URL for query parameters and return a
+table with the information for each query parameter.
+Example: http://localhost:5000/attributes?hello=world&lorem=ipsum
+hello : world
+lerem : ipsum
+Consider other test cases to check the robustness of your solution.
+- This implementation should work with a variety of query parameters.
+- Further Reading:
+- MDN - URL() interface https://developer.mozilla.org/en-US/docs/Web/API/URL
+*/
 const server = http.createServer((req, res) => {
   const routes = [
     '/attributes?hello=world&lorem=ipsum',
     '/items?first=1&second=2&third=3&fourth=4',
     '/characters?spongebob=squarepants&patrick=star&sandy=cheeks',
+    '/somethingelse?page=1&limit=5&productType=type1&productType=type2&productType=type3&brand=brand1&brand=brand2',
+    '/somethingelse2?page=&limit=&&',
   ];
 
   // use the URL interface to work with URLs
   // source: https://developer.mozilla.org/en-US/docs/Web/API/URL
-  let url = new URL(req.url, `http://${req.headers.host}`);
+  const url = new URL(req.url, `http://${req.headers.host}`);
 
-  let getRoutes = () => {
+  const getRoutes = () => {
     let result = '';
-
     routes.forEach(
-      (elem) => (result += `<li><a href="${elem}">${elem}</a></li>`)
+      (elem) => { result += `<li><a href="${elem}">${elem}</a></li>`; },
     );
-
     return result;
   };
-
+  const getKeyValParams = () => {
+    let result = '<ul>';
+    const entries = Array.from(url.searchParams.entries());
+    for (let i = 0; i < entries.length; i += 1) {
+      const key = entries[i][0];
+      const val = entries[i][1];
+      result += `<li>${key} : ${val}</li>`;
```

Comment:
I think the assignment specifically asked for a table? Maybe you can change the `<li></li>` tags for the table tags instead?

Reply:
that is true, thank you. will change